### ISSUE TITLE
Looking for a Maintainer

### ISSUE THREAD

**Issue by sklose** [sklose/NCalc2#61]
Unfortunately, I do not have enough time to spare to actively maintain this library and address all the open feature requests and bug reports. Therefore, I am looking for somebody interested in becoming a maintainer for this project. Please reply to this issue if you are interested.

**Comment by davidnmbond** [comment]
Hi there - we (https://github.com/panoramicdata/) are the maintainers of PanoramicData.NCalcExtensions (https://github.com/panoramicdata/PanoramicData.NCalcExtensions), which uses NCalc.NetCore (abandoned since 2017).  We love NCalc and are looking to secure its future!  There are various forks of NCalc, and we are still reviewing where we should hang our hat, or even whether to create our own fork.

Your library certainly gets regular commits and the performance work you have been doing looks great!  Please let us know how we can help...

David / Panoramic Data

**Comment by sklose** [comment]
Hi @davidnmbond 

thanks for reaching out! I think there are many ways how you could help out:

- [ ] there is a long list of open issues, that need addressing (either by implementing new features, fixing bugs, or deciding that it's something NCalc can't/won't/shouldn't support)
- [ ] the codebase could probably need a bit of an overhaul
    * the grammar file for generating the syntax parser isn't being used, ideally that generated code wouldn't be checked in, and instead is auto generates every time (I believe changes to the grammar file would be needed for some of the open issues)
    * there are subtle differences in behavior between interpreted execution and compiled lambdas, making the behavior exactly the same would be great
    * there might be some things in newer versions of .NET that the library could make use of (I haven't done any .NET development for the past 5 years, I am behind on what's available in newer versions)
    * hooking up code linting / code analysis tools into the CI pipeline
- [ ] some non-technical things
    * contribution guidelines / Issue and PR templates
    * improved README / documentation

My suggestions would be: if you are interested, start looking into any of these items (or anything else that you think is useful and I missed in the list). After you created a few PRs and everything looks good, I would give you Collaborator permissions to this repository and from then on you will be able to merge PRs yourself and manage the repo more autonomously.

Best,
Sebastian

**Comment by j4cegh** [comment]
> Hi there - we (https://github.com/panoramicdata/) are the maintainers of PanoramicData.NCalcExtensions (https://github.com/panoramicdata/PanoramicData.NCalcExtensions), which uses NCalc.NetCore (abandoned since 2017). We love NCalc and are looking to secure its future! There are various forks of NCalc, and we are still reviewing where we should hang our hat, or even whether to create our own fork.
> 
> Your library certainly gets regular commits and the performance work you have been doing looks great! Please let us know how we can help...
> 
> David / Panoramic Data

David, PLEASE help maintain this. It's an insane library! If you all could do an even better job on performance, a lot of people would surely appreciate it!

**Comment by Bykiev** [comment]
@david-brink-talogy, what do you think about became a maintainer? I see @sklose doesn't have enough time, but I want to make some more changes in NCalc and my last PR wasn't merge since the last week. I see you have all needed skills to review PRs and improve the library.

**Comment by sklose** [comment]
happy to grant you collaborator permissions @david-brink-talogy so you can approve PRs and things don't get stalled when I don't have time to respond. let me know if you want it.

**Comment by gumbarros** [comment]
I would love to be a maintainer, I created 5 PRs and no one responded me near a week at the original NCalc :(

**Comment by Bykiev** [comment]
@david-brink-talogy, 

> I would love to be a maintainer, I created 5 PRs and no one responded me near a week at the original NCalc :(
> 
> Also what about a rebranding to not be confused with the original NCalc? Like ExpressionsSharp or something.

Hi, welcome to contribute in this fork. Let's make NCalc super fast!

**Comment by Bykiev** [comment]
@gumbarros, I review your changes in NCalc, would you like to create a PR with cache changes here?

**Comment by david-brink-talogy** [comment]
> happy to grant you collaborator permissions @david-brink-talogy so you can approve PRs and things don't get stalled when I don't have time to respond. let me know if you want it.

That works for me, but I'd want to be clear that while I'm happy to review PRs and participate in discussion, my available time may be sporadic based on the needs of my day job.

**Comment by gumbarros** [comment]
> @gumbarros, I reviewed your changes in NCalc, would you like to create a PR with cache changes here? It should close https://github.com/sklose/NCalc2/issues/39

Yes, at the weekend I will re-do the changes here, in a hurry this week.
